### PR TITLE
Ros2 collision fix

### PR DESCRIPTION
**Pull request by ANGELOANTU7**
Description:

This pull request addresses issue https://github.com/blackcoffeerobotics/bcr_bot/issues/23. The issue was regarding misalignment of the wheels causing them to be out of the chassis in the simulation test. The changes made in this PR have corrected the alignment of the wheels, allowing them to move correctly within the chassis during simulation tests.

Changes Made:

Fixed misalignment of wheels.
Ensured wheels are movable within the chassis in simulation tests.

Checklist
Create tests, which fail without this PR.

[Screencast from 03-13-2024 04_34_11 PM.webm](https://github.com/blackcoffeerobotics/bcr_bot/assets/94218410/0fde341f-1789-43d7-9b8f-2c2ce4c164e5)
[Screencast from 03-13-2024 04_27_30 PM.webm](https://github.com/blackcoffeerobotics/bcr_bot/assets/94218410/7d704f74-e359-4fb8-acd9-2dee3fe9bb7f)

### CODE REVIEW

**Comment by D-1shu** [comment]
@ANGELOANTU7, could you please address the following points:

1. I am unable to observe the collisions being fixed Instead they are removed all together. I've followed these steps to verify the collisions. 
    a. Execute `ros2 launch bcr_bot gazebo.launch.py`.
    b. Execute `ros2 launch bcr_bot rviz.launch.py`.

2. After resolving the issue, squash the commits into a single commit and pull request should thoroughly explain the changes made. For future PRs kindly adhere to the instructions provided [here](https://gitlab.com/bcrllp/processes/development/-/blob/master/version-control-and-code-review.md?ref_type=heads) for version control and code review.

**Comment by ANGELOANTU7** [comment]
edit made

**Comment by D-1shu** [comment]
@ANGELOANTU7 Both these launch file results in error
a. `ros2 launch bcr_bot gazebo.launch.py`
b.  `ros2 launch bcr_bot gz.launch.py`
```
[ERROR] [launch]: Caught exception in launch (see debug for traceback): executed command failed. Command: xacro /home/devanshu/bcr_bot_antu/install/bcr_bot/share/bcr_bot/urdf/bcr_bot.xacro camera_enabled:=True stereo_camera_enabled:=False two_d_lidar_enabled:=True sim_gazebo:=true odometry_source:=world robot_namespace:=
Captured stderr output: XML parsing error: not well-formed (invalid token): line 52, column 1
when processing file: /home/devanshu/bcr_bot_antu/install/bcr_bot/share/bcr_bot/urdf/macros.xacro
included from: /home/devanshu/bcr_bot_antu/install/bcr_bot/share/bcr_bot/urdf/bcr_bot.xacro

Check that:
 - Your XML is well-formed
 - You have the xacro xmlns declaration: xmlns:xacro="http://www.ros.org/wiki/xacro"

```

**Comment by ANGELOANTU7** [comment]
resolved the issue

**Review by D-1shu — COMMENTED** [review]
LGTM!

File: urdf/bcr_bot.xacro [hunk 1]
```diff
@@ -103,8 +103,8 @@
     <xacro:trolley_wheel cardinality="back" dexterity="left" origin_x="-${chassis_length/2}" origin_y="${trolley_track_width/2}" origin_z="-${trolley_wheel_radius+chassis_height/2}"/>
     <xacro:trolley_wheel cardinality="back" dexterity="right" origin_x="-${chassis_length/2}" origin_y="-${trolley_track_width/2}" origin_z="-${trolley_wheel_radius+chassis_height/2}"/>
 
-    <xacro:traction_wheel cardinality="middle" dexterity="left" origin_x="0" origin_y="${traction_track_width/2}" origin_z="-${chassis_height/2+2*trolley_wheel_radius+eps-traction_wheel_radius}"/>
-    <xacro:traction_wheel cardinality="middle" dexterity="right" origin_x="0" origin_y="-${traction_track_width/2}" origin_z="-${chassis_height/2+2*trolley_wheel_radius+eps-traction_wheel_radius}"/>
+    <xacro:traction_wheel cardinality="middle" dexterity="left" origin_x="0" origin_y="${traction_track_width/2.75}" origin_z="-${chassis_height/2+2*trolley_wheel_radius+eps-traction_wheel_radius}"/>
```
Comment (ggupta9777) [inline urdf/bcr_bot.xacro:106]: @ANGELOANTU7  Why have you divided this by 2.75? This is an arbitrary choice

**Review by ggupta9777 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**